addons/mrp/models/mrp_production.py

```diff
@@ -1854,7 +1856,24 @@ def _default_amounts(production):
                     new_moves_vals.append(move_vals)
                     moves.append(move)
 
-        backorder_moves = self.env['stock.move'].create(new_moves_vals)
+        # Merge move vals for the same product from the same source if no explicit amounts were provided
+        updated_moves_vals = {}
+        if not amounts_arg:
+            for move_vals in new_moves_vals:
+                key_tuple = (move_vals['product_id'], move_vals['location_id'])
+                if key_tuple in updated_moves_vals:
+                    if 'quantity' in updated_moves_vals[key_tuple]:
+                        updated_moves_vals[key_tuple]['quantity'] += move_vals.get('quantity', 0)
```

Comment:
in my opinion, it would be easier to read if we keep the move_vals in a variable than to use tuple everywhere, eg. for the beginning:

```suggestion
            for move_vals in new_moves_vals:
                key_tuple = (move_vals['product_id'], move_vals['location_id'])
                target_vals = updated_move_vals.get(key_tuple)
                if target_vals:
                    if 'quantity' in merge_into_vals:
                        target_vals['quantity'] += move_vals.get('quantity', 0)
                    # ...
```
also we could use some get to reduce a little the number of lines, eg:

```diff
- if 'quantity' in updated_moves_vals[key_tuple]:
-     updated_moves_vals[key_tuple]['quantity'] += move_vals.get('quantity', 0)
- else:
-     updated_moves_vals[key_tuple]['quantity'] = move_vals.get('quantity', 0)
+ target_vals['quantity'] = target_vals.get('quantity', 0) + move_vals.get('quantity', 0)